org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/managers/ProjectsManager.java

```diff
@@ -88,11 +103,39 @@ public void fileChanged(String uriString, CHANGE_TYPE changeType) {
 			if (resource != null) {
 				resource.refreshLocal(IResource.DEPTH_INFINITE, new NullProgressMonitor());
 			}
+			if (isBuildFile(resource)) {
```

Comment:
If the build file is deleted this will just ignore it because resource has been changed to parent earlier. Should the deletion be explicitly handled?